spring-data-relational/src/main/java/org/springframework/data/relational/core/mapping/schemasqlgeneration/SchemaSQLGenerator.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.relational.core.mapping.schemasqlgeneration;
+
+import org.springframework.data.relational.core.sql.IdentifierProcessing;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class SchemaSQLGenerator {
+
+    private final IdentifierProcessing identifierProcssing;
+    public SchemaSQLGenerator(IdentifierProcessing identifierProcessing) {
```

Comment:
@schauder - 

Is there a way to get the IdentifierProcessing from the RelationalMappingContext?   (or when user's use this class to generate SQL - that they can easily get an IdentifierProcessing object - that matches the one they are using for executing SQL)?

For the Unit Test(s) I think passing it in gives flexibility.